lms/static/scripts/frontend_apps/components/GroupConfigSelector.js

```diff
@@ -52,21 +54,30 @@ export default function GroupConfigSelector({
   groupConfig,
   onChangeGroupConfig,
 }) {
-  const [groupSets, setGroupSets] = useState(
-    /** @type {GroupSet[]|null} */ (null)
-  );
-
   const [fetchError, setFetchError] = useState(
     /** @type {Error|null} */ (null)
   );
 
+  const [groupSets, setGroupSets] = useState(
+    /** @type {GroupSet[]|null} */ (null)
+  );
+
   const {
     api: { authToken },
     filePicker: { blackboard, canvas },
   } = useContext(Config);
 
+  const useGroupSet = groupConfig.useGroupSet;
+  const haveFetchedGroups = groupSets !== null;
+
+  const groupSet = useGroupSet ? groupConfig.groupSet : null;
+  const fetchingGroupSets = !groupSets && !fetchError && useGroupSet;
+
```

Comment:
Some of the diff here is me moving reference assignments around as there are a lot of them and I was attempting to group and re-order a little more clearly.